src/pages/settings/addressbook/addressbook.ts

```diff
@@ -54,6 +70,33 @@ export class AddressbookPage {
       });
   }
 
+  public reorder(): void {
+    this.showReorder = !this.showReorder;
+  }
+
+  public setAddressOrder(address: string, index: number): void {
+    this.addressbookProvider.setAddressOrder(address, index).then(() => {
+      this.logger.debug(
+        'Address new order stored for ' + address + ': ' + index
+      );
+    });
+    if (this.addressbook[address]) this.addressbook[address]['order'] = index;
+  }
+
+  public async getAddressOrder(address: string): Promise<any> {
+    return this.addressbookProvider.getAddressOrder(address);
+  }
+
+  public reorderAddresses(indexes): void {
+    const element = this.addressbook[indexes.from];
+    this.addressbook.splice(indexes.from, 1);
+    this.addressbook.splice(indexes.to, 0, element);
+    _.each(this.addressbook, (entry, index: number) => {
+      entry['order'] = index;
```

Comment:
this is not needed

Reply:
Without setting entry['order'] to the current index, the reordering functionality for each sliding component goes back back to the original index.